core/src/main/scala/com/evolutiongaming/kafka/flow/kafka/KafkaModule.scala

```diff
@@ -1,83 +1,80 @@
 package com.evolutiongaming.kafka.flow.kafka
 
-import cats.effect.Blocker
-import cats.effect.Clock
-import cats.effect.ConcurrentEffect
-import cats.effect.ContextShift
-import cats.effect.Resource
-import cats.effect.Timer
+import cats.effect.{Blocker, Clock, ConcurrentEffect, ContextShift, Resource, Timer}
 import cats.syntax.all._
-import com.evolutiongaming.catshelper.FromTry
-import com.evolutiongaming.catshelper.LogOf
-import com.evolutiongaming.catshelper.ToFuture
-import com.evolutiongaming.catshelper.ToTry
+import com.evolutiongaming.catshelper.CatsHelper._
+import com.evolutiongaming.catshelper.{FromTry, LogOf, ToFuture, ToTry}
 import com.evolutiongaming.kafka.flow.LogResource
-import com.evolutiongaming.kafka.journal.KafkaConfig
-import com.evolutiongaming.kafka.journal.KafkaHealthCheck
-import com.evolutiongaming.kafka.journal.RandomIdOf
 import com.evolutiongaming.kafka.journal.util.SkafkaHelper._
-import com.evolutiongaming.kafka.journal.{KafkaConsumerOf => JournalConsumerOf}
-import com.evolutiongaming.kafka.journal.{KafkaProducerOf => JournalProducerOf}
-import com.evolutiongaming.skafka.consumer.AutoOffsetReset
-import com.evolutiongaming.skafka.consumer.{ConsumerConfig, ConsumerMetrics, ConsumerOf => RawConsumerOf}
-import com.evolutiongaming.skafka.producer.{ProducerConfig, ProducerMetrics, ProducerOf => RawProducerOf}
-import com.evolutiongaming.smetrics.CollectorRegistry
-import com.evolutiongaming.smetrics.MeasureDuration
+import com.evolutiongaming.kafka.journal.{
+  KafkaConfig,
+  KafkaHealthCheck,
+  RandomIdOf,
+  KafkaConsumerOf => JournalConsumerOf,
+  KafkaProducerOf => JournalProducerOf
+}
+import com.evolutiongaming.skafka.consumer._
+import com.evolutiongaming.skafka.producer.{Producer, ProducerMetrics, ProducerOf}
+import com.evolutiongaming.smetrics.{CollectorRegistry, MeasureDuration}
+import monocle.macros.syntax.lens._
 import scodec.bits.ByteVector
 
 trait KafkaModule[F[_]] {
 
   def healthCheck: KafkaHealthCheck[F]
 
-  def consumerOf: ConsumerOf[F]
-  def producerOf: RawProducerOf[F]
+  def consumerOf: Resource[F, Consumer[F, String, ByteVector]]
+  def producerOf: Resource[F, Producer[F]]
 
 }
 object KafkaModule {
 
   def of[F[_]: ConcurrentEffect: ContextShift: FromTry: ToTry: ToFuture: Timer: LogOf](
     applicationId: String,
-    config: ConsumerConfig,
+    groupId: String,
```

Comment:
I think the reason that `ConsumerOf` was introduced is that one application might want to read one and the same topic several times for different purposes with different consumer group.

Reply:
That's bit unexpected usage, could you provide an example?

Reply:
Defining two distinct `KafkaFlow.resource` resources in one application.

To give an example, one might want to read game journal topic both for sending DWH events and for generating game stats for purpose of showing these to players.

Reply:
Then one should define two `KafkaModule` in the application that will run under the hood two Kafka consumers.

I think passing configuration somewhere after creation of `KafkaModule` is wrong approach that makes clients pass the configuration deeper throw multiple layers.

Reply:
In this case you will get two health checks etc., which seems to be a no-go.

Reply:
Why not? Two different Kafka clients should have independent health check as far as they can fail independently.

Reply:
We will have 3-4 clients then:
1. Client for consuming journals.
2. Client for producing messages for the applications where it is required.
3. Client for producing and consuming health checks.
4. Client for producing and consuming health checks.

So, (3) and (4) will do exactly the same, but somehow confirm the independence of (1) and (2), or I misunderstood the logic I see in the code?

Reply:
That's why one should not run multiple Kafka consumers in one JVM. Instead one should subscribe on multiple topics from the same consumer in case of using same brokers. 

For rare cases of using different brokers having independent health-check doesn't looks like an issue.

Reply:
> That's why one should not run multiple Kafka consumers in one JVM. Instead one should subscribe on multiple topics from the same consumer in case of using same brokers.
What if we want to use the same (or different?) topic, but different `groupId`? Create new JVM instance for every `groupId`? It does not sound like a great idea. It will decrease microservice density a lot for no specific reason...

I think we wandered away from the topic though. The reason I have added `KafkaModule` at all, because I was trying to move as much boilerplate as possible from the user to the library. The original library only had `ConsumerOf` and user had to configure it himself.

The whole purpose of `KafkaModule` was "consumer + health check". Should we just remove healthcheck from `kafka-flow` altogether and `KafkaModule` as a consequence? :thinking: